src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasService.kt

```diff
@@ -0,0 +1,76 @@
+/**
+Copyright 2018 BarD Software s.r.o
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+package com.bardsoftware.papeeria.backend.cosmas
+
+import com.google.protobuf.ByteString
+import io.grpc.stub.StreamObserver
+
+/**
+ * Special class that can work with requests from client
+ */
+class CosmasService : CosmasGrpc.CosmasImplBase() {
+
+    private val files = mutableMapOf<String, MutableList<ByteString>>()
+
+    override fun getVersion(request: CosmasProto.GetVersionRequest,
+                            responseObserver: StreamObserver<CosmasProto.GetVersionResponse>) {
+        println("Get request for version: ${request.version}")
+        val response = CosmasProto.GetVersionResponse.newBuilder()
+        synchronized(files) {
+            val fileVersions = files[request.fileId]
+            if (fileVersions != null &&

```

Comment:
One more tip how to make the code smaller and avoid duplicating yourself: rename `printErrorInRequest` to `verifyGetVersion(fileVersions, request): Error` and call it like
```
val error = verifyGetVersion(fileVersions, request)
if (error == null) {
   // get file and build
} else {
  // do error handling the way you like (see below), use fields from returned error to construct appropriate response or send error to the client
}

Reply:
Ok, but kotlin doesn't understand that if (error == null) then fileVersions != null, so i have to write something like `fileVersions?.get(request.version)` or `fileVersions!![requset.version]`